test/devicecontroller/DefaultDeviceController.test.ts

```diff
@@ -2398,28 +2398,102 @@ describe('DefaultDeviceController', () => {
+      expect(element.srcObject === (await gUM.returnValues.pop()));
+      expect(spy1.notCalled).to.be.true;
+      expect(spy2.calledTwice).to.be.true;
+      expect(gUM.calledOnce).to.be.true;
+      spy1.restore();
+      spy2.restore();
+      gUM.restore();
+    });
+
+    it('connects the active device video stream to the video element if there is no transform device stream', async () => {
+      const spy1 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
+      const spy2 = sinon.spy(DefaultVideoTile, 'connectVideoStreamToVideoElement');
+      const gUM = sinon.spy(navigator.mediaDevices, 'getUserMedia');
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+
+      deviceController.startVideoPreviewForVideoInput(element);
+
+      expect(gUM.returnValues).to.have.lengthOf(1);
+      expect(element.srcObject === (await gUM.returnValues.pop()));
+      expect(spy1.notCalled).to.be.true;
       expect(spy2.calledOnce).to.be.true;
       expect(gUM.calledOnce).to.be.true;
       spy1.restore();
       spy2.restore();
       gUM.restore();
     });
 
-    it('does not disconnect or connect the video stream for preview if no active video exists', () => {
+    it('reuses the original video stream to create a transform device stream', async () => {
       const spy1 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
       const spy2 = sinon.spy(DefaultVideoTile, 'connectVideoStreamToVideoElement');
       const gUM = sinon.spy(navigator.mediaDevices, 'getUserMedia');
+      domMockBehavior.createElementCaptureStream = new MediaStream();
+      domMockBehavior.mediaStreamTrackSettings.deviceId = stringDeviceId;
+      domMockBehavior.createElementCaptureStream.addTrack(new MediaStreamTrack());
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+      deviceController.startVideoPreviewForVideoInput(element);
+      const processor = new NoOpVideoFrameProcessor();
+      const device = new DefaultVideoTransformDevice(logger, stringDeviceId, [processor]);
+      await deviceController.chooseVideoInputDevice(device);
+      deviceController.startVideoPreviewForVideoInput(element);
+
+      expect(gUM.returnValues).to.have.lengthOf(1);
+      expect(spy1.notCalled).to.be.true;
+      expect(spy2.calledTwice).to.be.true;
+      expect(gUM.calledOnce).to.be.true;
+      spy1.restore();
+      spy2.restore();
+      gUM.restore();
+    });
+
+    it('disconnects and releases the video stream if the video element stream is different from the active video stream', async () => {
+      await deviceController.chooseVideoInputDevice(stringDeviceId);
+      deviceController.startVideoPreviewForVideoInput(element);
+      const newStringDeviceId = 'device-id-with-transformation';
+      const spy1 = sinon.spy(deviceController, 'releaseMediaStream');
+      const spy2 = sinon.spy(DefaultVideoTile, 'disconnectVideoStreamFromVideoElement');
```

Comment:
Do we need to check for `connectVideoStreamToVideoElement` also?

Reply:
I think we should. I will add that expect statement in my update.